gvpn/TincanDispatcher.py

```diff
@@ -48,104 +48,118 @@ def processCBT(self, cbt):
+
+                elif msg_type == "local_state":
+
+                    # Send CBT to Watchdog to store ipop_state
+
+                    CBT = self.CFxHandle.createCBT(initiator='TincanDispatcher',
+                                                   recipient='Watchdog',
+                                                   action='STORE_IPOP_STATE',
+                                                   data=msg)
+                    self.CFxHandle.submitCBT(CBT)
+
+                elif msg_type == "peer_state":
+
+                    # Send CBT to Monitor to store peer state
+
+                    CBT = self.CFxHandle.createCBT(initiator='TincanDispatcher',
+                                                   recipient='Monitor',
+                                                   action='STORE_PEER_STATE',
+                                                   data=msg)
+                    self.CFxHandle.submitCBT(CBT)
+
+                elif (msg_type == "con_stat" or msg_type == "con_req" or
+                      msg_type == "con_resp" or msg_type == "send_msg"):
+
+                    CBT = self.CFxHandle.createCBT(initiator='TincanDispatcher',
+                                                   recipient='BaseTopologyManager',
+                                                   action='TINCAN_MSG',
+                                                   data=msg)
+                    self.CFxHandle.submitCBT(CBT)
+
+            #  If a packet that is destined to yet no p2p connection
+            #  established node, the packet as a whole is forwarded to
+            #  controller
+            # |-------------------------------------------------------------|
+            # | offset(byte) |                                              |
+            # |-------------------------------------------------------------|
+            # |      0       | ipop version                                 |
+            # |      1       | message type                                 |
+            # |      2       | source uid                                   |
+            # |     22       | destination uid                              |
+            # |     42       | Payload (Ethernet frame)                     |
+            # |-------------------------------------------------------------|
+
+            elif data[1] == self.tincan_packet:
+
+                # Send the Tincan Packet to BaseTopologyManager
+
+                packet = data[2:]
                 CBT = self.CFxHandle.createCBT(initiator='TincanDispatcher',
-                                               recipient='Watchdog',
-                                               action='STORE_IPOP_STATE',
-                                               data=msg)
+                                               recipient='BaseTopologyManager',
+                                               action='TINCAN_PACKET',
+                                               data=packet)
```

Comment:
I agree with you, but in line [264](https://github.com/GingerNinja23/controller-framework/blob/master/gvpn/BaseTopologyManager.py#L264) of BaseTopologyManager, we are logging the IPOP version and message type also.